src/transaction.rs

```diff
@@ -70,6 +63,37 @@ impl Default for ResolveCondition {
     }
 }
 
+/// Represents a prepared and optionally signed transaction that is ready for
+/// sending created by a `TransactionBuilder`.
+#[derive(Clone, Debug, PartialEq)]
+#[allow(clippy::large_enum_variant)]
+pub enum Transaction {
+    /// A structured transaction request to be signed locally by the node.
+    Request(TransactionRequest),
+    /// A signed raw transaction request.
+    Raw(Bytes),
+}
+
+impl Transaction {
+    /// Unwraps the transaction into a transaction request, returning None if the
+    /// transaction is a raw transaction.
```

Comment:
Don't all successful transactions return a receipt of some type (regardless of whether we were expecting a return value).

Reply:
This is for sending transactions, they can either be a transaction object (which is unsigned and requires the node to sign transactions locally) or a raw transactions (a bunch of bytes, that are signed). This utility method just returns `Some(TransactionRequest)` if it is the former, and `None` if it is the latter.